spec/services/job_processor_spec.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+class TestJob < ActiveJob::Base
+  def initialize
+    @file = Tempfile.new("test-job-result")
+    super
+  end
+
+  def perform(message)
+    @file.write(message)
+  end
+
+  def result
+    @file.rewind
+    @file.read
+  end
+end
+
+describe JobProcessor do
+  describe ".perform_forked" do
+    let(:job) { TestJob.new }
+
+    it "executes a job" do
+      JobProcessor.perform_forked(job, "hello")
+
+      expect(job.result).to eq "hello"
```

Comment:
Actually, I'm wondering if it's necessary to mock up the TestJob at all. Couldn't we just create a double and
```suggestion
      expect(job).to receive(:perform).with("hello")
```
?

I guess this way specifies in more detail how this class is intended to be used.

Reply:
I see doubles as backup option if it's too complicated to use real classes. Tests with real classes are usually more robust and test closer to production code.

And I think that your example wouldn't work. The forked process would call `perform` and our test process wouldn't know about it. Our double wouldn't know that the cloned/forked double was called. The only way to communicate between the two processes is via a file (or socket etc).

Reply:
Ah, good points, thanks for explaining that.